### PR TITLE
Disable private feed sharing

### PR DESCRIPTION
**Pull request by bjtitus**
Disables sharing of private feeds with a Toast message shown when trying to share.

The core feature is flagged under `disable_private_feed_sharing` but I left the database changes unflagged.

https://github.com/user-attachments/assets/ea3ecd31-e997-49c3-aa71-5db3929743d5

## To test

1. Try sharing a podcast and make sure it works
2. Try sharing a private podcast (try this URL: `https://pca.st/private/707a8bf0-9abf-013c-74fe-027201e7e97f`)
3. Ensure this message appears in a toast: "Sharing is not available for private podcasts"

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: podcasts/Sharing/SharingModal.swift [hunk 1]
```diff
@@ -101,6 +101,16 @@ enum SharingModal {
     }
 
     static func show(option: Option, from source: AnalyticsSource, in viewController: UIViewController) {
+
+        if FeatureFlag.disablePrivateFeedSharing.enabled {
+            if case let .podcast(podcast) = option {
+                guard !podcast.isPrivate else {
```
Comment (danielebogo) [inline podcasts/Sharing/SharingModal.swift]: Nit: in this case `if podcast.isPrivate` is more readable

File: podcasts/SharingHelper.swift [hunk 2]
```diff
@@ -7,6 +7,14 @@ class SharingHelper: NSObject {
     static let shared = SharingHelper()
     var activityController: UIActivityViewController?
     func shareLinkTo(podcast: Podcast, fromController: UIViewController, fromSource: AnalyticsSource, sourceRect: CGRect, sourceView: UIView) {
+
+        if FeatureFlag.disablePrivateFeedSharing.enabled {
+            guard !podcast.isPrivate else {
```
Comment (danielebogo) [inline podcasts/SharingHelper.swift:12]: Nit: in this case `if podcast.isPrivate` is more readable

**Review by danielebogo — CHANGES_REQUESTED** [review]
Hey @bjtitus I left few nit.
The disabled share worked on the podcast page and full screen player.
however I'm still able to share the episode, current position and clip for a specific episode. Is this expected?

https://github.com/user-attachments/assets/01cfb7bc-51e7-4e90-9a2f-9965438f3dfe

File: podcasts/Sharing/SharingModal.swift (hunk 1 above)
Comment (bjtitus) [reply podcasts/Sharing/SharingModal.swift]: Kind of prefer the `guard` for the early returns but it is more concise here with the other if statements.

**Comment by bjtitus** [comment]
@danielebogo Should be all fixed up now.

**Review by danielebogo — APPROVED** [review]
:shipit:

**Pull request merged**